src/Featurizers/AnalyticalRollingWindowFeaturizer.h

```diff
@@ -0,0 +1,337 @@
+            std::uint32_t                       horizon(Traits<std::uint32_t>::deserialize(ar));
+            std::uint32_t                       minWindowSize(Traits<std::uint32_t>::deserialize(ar));
+
+
+            return AnalyticalRollingWindowTransformer(std::move(maxWindowSize), std::move(windowCalculation), std::move(horizon), std::move(minWindowSize));
+        }()
+    ) {
+}
+
+template <typename InputT, size_t MaxNumTrainingItemsV>
+bool AnalyticalRollingWindowTransformer<InputT, MaxNumTrainingItemsV>::operator==(AnalyticalRollingWindowTransformer const &other) const {
+    return this->_maxWindowSize  == other._maxWindowSize && this->_windowCalculation == other._windowCalculation && this->_horizon == other._horizon && this->_minWindowSize == other._minWindowSize;
+}
+
+template <typename InputT, size_t MaxNumTrainingItemsV>
+bool AnalyticalRollingWindowTransformer<InputT, MaxNumTrainingItemsV>::operator!=(AnalyticalRollingWindowTransformer const &other) const {
+    return (*this == other) == false;
+}
+
+template <typename InputT, size_t MaxNumTrainingItemsV>
+void AnalyticalRollingWindowTransformer<InputT, MaxNumTrainingItemsV>::save(Archive &ar) const /*override*/ {
+    // Version
+    Traits<std::uint16_t>::serialize(ar, 1); // Major
+    Traits<std::uint16_t>::serialize(ar, 0); // Minor
+
+    // Data
+    Traits<std::uint32_t>::serialize(ar, this->_maxWindowSize);
+    Traits<std::uint8_t>::serialize(ar, static_cast<std::uint8_t>(_windowCalculation));
+    Traits<std::uint32_t>::serialize(ar, this->_horizon);
+    Traits<std::uint32_t>::serialize(ar, this->_minWindowSize);
+
+    // Note that we aren't serializing working state
+}
+
+// ----------------------------------------------------------------------
+// |
+// |  AnalyticalRollingWindowEstimator
+// |
+// ----------------------------------------------------------------------
+template <typename InputT, size_t MaxNumTrainingItemsV>
+AnalyticalRollingWindowEstimator<InputT, MaxNumTrainingItemsV>::AnalyticalRollingWindowEstimator(AnnotationMapsPtr pAllColumnAnnotations, std::uint32_t maxWindowSize, AnalyticalRollingWindowCalculation windowCalculation, std::uint32_t horizon, std::uint32_t minWindowSize) :
+    BaseType(AnalyticalRollingWindowEstimatorName, std::move(pAllColumnAnnotations)),
+    _maxWindowSize(
+        std::move(
+            [&maxWindowSize]() -> std::uint32_t & {
+                if(maxWindowSize < 1)
+                    throw std::invalid_argument("maxWindowSize");
+
+                return maxWindowSize;
+            }()
+        )
+    ),
+     _horizon(
+        std::move(
+            [&horizon]() -> std::uint32_t & {
+                if(horizon < 1)
+                    throw std::invalid_argument("horizon");
+
+                return horizon;
+            }()
+        )
+    ),
+    _minWindowSize(
+        std::move(
+            [&minWindowSize]() -> std::uint32_t & {

```

Comment:
>minWindowSize [](start = 14, length = 13)

Ensure that `_minWindowSize` is < `_maxWindowSize`.

Reply:
Check has been added.